packages/react-app/src/components/pages/Bounties/Form/NewBounty.tsx

```diff
@@ -0,0 +1,249 @@
+import { Controller, useForm } from 'react-hook-form';
+import React, { useContext } from 'react';
+import { useRouter } from 'next/router';
+import DatePicker from '@app/components/parts/DatePicker';
+import {
+	Box,
+	FormLabel,
+	FormControl,
+	Input,
+	Button,
+	FormErrorMessage,
+	Select,
+	Flex,
+	FormControlProps,
+	Textarea,
+} from '@chakra-ui/react';
+import { BountyCollection } from '@app/models/Bounty';
+import bountyStatus from '@app/constants/bountyStatus';
+import { CustomerContext } from '@app/context/CustomerContext';
+import { APIUser } from 'discord-api-types';
+import { useUser } from '@app/hooks/useUser';
+import { useLocalStorage } from '@app/hooks/useLocalStorage';
+import { dateIsNotInPast, required, validNonNegativeDecimal } from '@app/utils/formUtils';
+
+const PLACEHOLDERS = {
+	TITLE: 'Example: Create new Logo',
+	DESCRIPTION: 'Example: We need someone to create some snappy looking logos for our new Web3 project.',
+	CRITERIA: 'Example: SVG and PNG images approved by the team',
+};
+
+const useCurrencies = (): string[] => {
+	return ['BANK'];
+};
+
+const defaultValues = {
+	title: '',
+	description: '',
+	reward: '1000',
+	currency: 'BANK',
+	criteria: '',
+	dueAt: new Date().toISOString(),
+	gatedTo: '',
+	multiClaim: false,
+	evergreen: false,
+};
+
+const NotNeededFields = [
+	'_id',
+	'claimedBy',
+	'submissionNotes',
+	'submissionUrl',
+	'season',
+	'claimedAt',
+	'submittedAt',
+	'reviewedAt',
+	'reviewedBy',
+	'submittedBy',
+] as const;
+
+const useCachedForm = () => {
+	const { data: cachedBounty } = useLocalStorage<typeof defaultValues>('cachedEdit');
+	return cachedBounty ?? defaultValues;
+};
+
+const generatePreviewData = (
+	data: typeof defaultValues,
+	customer_id: string,
+	user: APIUser
+): Omit<BountyCollection, typeof NotNeededFields[number]> => {
```

Comment:
Re: `Omit` , Fair enough, but as bounty collection continues to add more fields, this will generate type errors. Might be easier to just to specify what fields are in the preview. I do like that the signature is a lot more readable however

Reply:
I disagree, will only add type errors if you need to add fields that are in "NotNeededFields", you as a developer can very easily change.